jetty-setuid/jetty-setuid-jna/src/main/java/org/eclipse/jetty/setuid/LibC.java

```diff
@@ -0,0 +1,65 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.setuid;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.Platform;
+
+/**
+ * Class is for changing user and groupId, it can also be use to retrieve user information by using getpwuid(uid) or getpwnam(username) of both linux and unix
+ * systems
+ */
+public interface LibC extends Library
+{
+    LibC INSTANCE = Native.load("c", LibC.class);
+
+    int setumask(int mask);
+
+    int setuid(int uid);
+
+    int setgid(int gid);
+
+    int setgroups(int[] gids);
+
+    Passwd getpwnam(String name);
+
+    Passwd getpwuid(int uid);
+
+    Group getgrnam(String name);
+
+    Group getgrgid(int gid);
+
+    int getrlimit(int resource, RLimit rLimit);
+
+    int setrlimit(int resource, RLimit rlimit);
+
+    class Constants
+    {
+        public static final int RLIMIT_NOFILE;
+        static
```

Comment:
Note: this is the only piece of platform-dependent code we have: we need to know the actual value of the C `RLIMIT_NOFILE` preprocessor constant that isn't standardized, so some googling/experimentation is required to fill up this collection of `if / else` statements.

This only impacts the file limit feature, the other features truly are fully platform-independent.